src/CattyTests/ViewController/HelpWebViewControllerTests.swift

```diff
@@ -0,0 +1,79 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+import WebKit
+import XCTest
+
+@testable import Pocket_Code
+
+class DemoNavigationAction: WKNavigationAction {
+    let testRequest: URLRequest
+    override var request: URLRequest {
+         testRequest
+    }
+
+    init(testRequest: URLRequest) {
+        self.testRequest = testRequest
+        super.init()
+    }
+}
+
+final class HelpWebViewControllerTests: XCTestCase {
+
+    func testDecidePolicyForNavigation() {
+        let webView = WKWebView()
+        let viewController = HelpWebViewController()
+
+        if let url = URL(string: "https://web-test.catrob.at/pocketcode/help") {
+            let testRequest1 = URLRequest(url: url)
+            let navigationAction = DemoNavigationAction(testRequest: testRequest1)
+            viewController.webView(webView, decidePolicyFor: navigationAction) { navigationActionPolicy in
+                XCTAssertEqual(navigationActionPolicy, WKNavigationActionPolicy.allow)
```

Comment:
Good approach with your `DemoNavigationAction`. However, when the _decisionHandler_ block is never called in the system under test, this test will still succeed. This is not good. You can use expectations to do this (have a look at `StoreProjectsDownloaderTests`) -- this has also the big advantage that the completion block can be dispatched on an asynchronous queue (currently this would not be possible).